### PR TITLE
fix(argo-cd): Also add hostAliases to dex deployment

### PR DESCRIPTION
**Pull request by Sinhyeok**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/dex/deployment.yaml [hunk 1]
```diff
@@ -127,6 +127,10 @@ spec:
       {{- end }}
     {{- end }}
       serviceAccountName: {{ template "argo-cd.dexServiceAccountName" . }}
+{{- with .Values.global.hostAliases }}
+      hostAliases:
+{{ toYaml . | indent 6 }}
+{{- end }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/dex/deployment.yaml:133]:
```suggestion
      {{- with .Values.global.hostAliases }}
      hostAliases:
        {{ toYaml . | indent 6 }}
      {{- end }}
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

**Review by mkilchhofer — APPROVED** [review]
Lgtm

**Review by mkilchhofer — APPROVED** [review]

**Comment by mkilchhofer** [comment]
Thank you so much for your contribution @Sinhyeok 🙏. I updated a few minor things (mostly regarding existing code style).

**Comment by Sinhyeok** [comment]
> Thank you so much for your contribution @Sinhyeok 🙏. I updated a few minor things (mostly regarding existing code style).

Thank you for reviewing :)

**Pull request merged**